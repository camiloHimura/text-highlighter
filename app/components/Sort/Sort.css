.sort{
    display: flex;
}

.sort .triangle{
    width: 0;
    height: 0;
    cursor: pointer;
    margin-left: 10px;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    border-top: 14px solid #555;
    transition: all .2s;
}

.sort .triangle.rotate{
    transform: rotate(180deg);
}
