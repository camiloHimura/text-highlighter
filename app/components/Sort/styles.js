import styled from 'styled-components'


export const Triangle = styled.span`
  width: 0;
  height: 0;
  cursor: pointer;
  margin-left: 10px;
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;
  border-top: 14px solid #555;
  transition: all .2s;

  &.rotate{
    transform: rotate(180deg);
  }
}
`;
