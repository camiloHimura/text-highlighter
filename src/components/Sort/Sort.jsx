import React from 'react';
import {connect} from 'react-redux';
import "./Sort.css"

import {sortAscendingTextAction} from "../../state/actions";

const mapStateToProps = state => {
    return {
            sortAscending: state.textList.sortAscending,
        }
}

const mapDispachToProps = dispatch => {
    return {
        sortAscendingText: info => dispatch(sortAscendingTextAction(info)),
    }
}


function Sort(props){
    const {sortAscendingText, sortAscending} = props;
    const className = sortAscending? "triangle rotate": "triangle";
    
    function changeSort(){
        sortAscendingText(!sortAscending);
    }

    return <div className="sort" onClick={changeSort}>
                Sort <span className={className}></span>
            </div>
}

export default connect(mapStateToProps, mapDispachToProps)(Sort);