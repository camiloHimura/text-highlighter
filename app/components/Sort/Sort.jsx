import React from 'react';
import {connect} from 'react-redux';
import PropTypes from "prop-types";
import {Triangle} from "./styles.js"

import {sortAscendingTextAction} from "../../state/actions";

const mapStateToProps = state => {
    return {
            sortAscending: state.textList.sortAscending,
        }
}

const mapDispachToProps = dispatch => {
  return {
    sortAscendingText: info => dispatch(sortAscendingTextAction(info)),
  }
}


function Sort(props){
  const {sortAscendingText, sortAscending} = props;
  
  function changeSort(){
    sortAscendingText(!sortAscending);
  }

  return <div className="sort --flex" onClick={changeSort}>
            Sort <Triangle className={`${sortAscending? "rotate": ""}`}></Triangle>
          </div>
}

Sort.propTypes = {
  sortAscending: PropTypes.bool.isRequired,
  sortAscendingText: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispachToProps)(Sort);
