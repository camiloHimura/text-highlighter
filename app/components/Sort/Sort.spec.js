import React from 'react';
import { Sort } from './Sort';
import { shallow } from 'enzyme'


describe('Colors component', () => {
  let Component;
  let props;

  beforeEach(() => {
    props = {
      sortAscending: false, 
      sortAscendingText: jest.fn(),
    }
    Component = shallow(<Sort {...props}/>);
  })
  it('take snapshot', () => {
    expect(Component).toMatchSnapshot();
  });

  it('changing sort value', () => {
    Component.find('div').simulate('click');
    expect(props.sortAscendingText).toHaveBeenCalledWith(!props.sortAscending);
  });


})