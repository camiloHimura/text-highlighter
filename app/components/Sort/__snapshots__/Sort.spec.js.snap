// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component take snapshot 1`] = `
<div
  className="sort --flex"
  onClick={[Function]}
>
  Sort 
  <styled.span
    className=""
  />
</div>
`;
